### PR TITLE
Improve CSScomb support

### PR DESCRIPTION
**Pull request by danielbayley**
Add setting for custom config file path and support for CSON in addition to the standard JSON config.

Signed-off-by: Daniel Bayley <daniel.bayley@me.com>

### CODE REVIEW

**Comment by danielbayley** [comment]
Following on from #209…

**Comment by danielbayley** [comment]
@Glavin001 Not sure why this is failing? I don't think it has anything to do with the code I added…

File: src/beautifiers/csscomb.coffee [hunk 1]
```diff
@@ -6,10 +6,10 @@ module.exports = class JSBeautify extends Beautifier
 
   options: {
     # TODO: Add support for options
-    CSS: false
-    LESS: false
-    Sass: false
-    SCSS: false
+    CSS: true
```
Comment (Glavin001) [inline src/beautifiers/csscomb.coffee]: Change:

```
    CSS: true
    LESS: true
    Sass: true
    SCSS: true
```

to:

```
    _:
      configPath: true
      predefinedConfig: true
    CSS: true
    LESS: true
    Sass: true
    SCSS: true
```

Similar to https://github.com/danielbayley/atom-beautify/blob/master/src/beautifiers/prettydiff.coffee#L6

This allows only the options that are supported (`configPath` and `predefinedConfig`) and puts that for the default (`_`).

File: src/beautifiers/csscomb.coffee [hunk 2]
```diff
@@ -18,16 +18,25 @@ module.exports = class JSBeautify extends Beautifier
 
       # Require
       Comb = require('csscomb')
+      expandHomeDir = require('expand-home-dir')
+      CSON = require('season')
 
+      readConfig = (configFile) ->
+        try
+          CSON.readFileSync(configFile)
```
Comment (Glavin001) [inline src/beautifiers/csscomb.coffee]: Does this also work for `.json` files? Has this been tested?

**Comment by Glavin001** [comment]
I usually ignore passing AppVeyor since Windows can be a PITA.

It is currently passing Travis CI, which is what I require before merging Pull Requests, so this is good :+1:.

After you fix the above recommended changes, please run `coffee docs/` as well which will generate the documentation with the new options you have added.

This is great. Good work so far! Will be able to merge this soon. Thanks.

**Comment by danielbayley** [comment]
`CSON.readFileSync` seems to actually read JSON too; so Iv'e simplified the logic to take that into account…

> since Windows can be a PITA.

*Can* be?!

File: src/beautifiers/csscomb.coffee (hunk 1 above)
Comment (Glavin001) [inline src/beautifiers/csscomb.coffee]: This is the last thing that needs fixing. Right now it looks like all options are available, when it should be only `configPath` and `predefinedConfig` option. After that, re-run `coffee docs/` to generate the correct documentation with only the allowed options. Thanks!

**Comment by Glavin001** [comment]
> CSON.readFileSync seems to actually read JSON too; so Iv'e simplified the logic to take that into account…

Great!

**Comment by danielbayley** [comment]
All sorted now as far as I can tell. The CI build is failing but since I'm getting deprecation warnings that include this package (since this last update to Atom itself) I'm guessing it's something to do with that…

**Comment by Glavin001** [comment]
It's actually failing because of the Titanium Style Sheets tests.
I'm going to merge this and take a look on my own.

**Comment by Glavin001** [comment]
Thank you!

**Comment by Glavin001** [comment]
Published to v0.28.17

**Pull request merged**